logger/logger.swift

```diff
@@ -1,10 +1,188 @@
 //
 //  logger.swift
-//  logger
+//  Test
 //
-//  Created by Daniel Brain on 10/12/20.
+//  Created by Daniel Brain on 10/11/20.
 //
 
-class logger {
+import Foundation
 
+let DEFAULT_PAYLOAD: Payload = [:]
+
+class Logger {
```

Comment:
We can also make this class a single instance, using the following:

```swift
public class Logger {
  static let shared: Logger = Logger()
}
```

And since we would use `OperationQueue` and our own underlying thread `loggingQueue` this would make this single instance thread safe.

We can of-course still allow people to init through:

```swift
    init(
        url : String,
        prefix : String? = nil,
        logLevel : LogLevel = DEFAULT_LOGLEVEL,
        flushInterval: Double = DEFAULT_FLUSH_INTERVAL
) {  
  // ..
}
```

Note if we go this route, we need to ensure the consumer sets the `url`, so allow them to change it via static func such as:

```swift
Logger.shared.setRequestUrl(myUrl)
```

Reply:
The only problem then is, if multiple people use the same shared logger and change the url we're in trouble, right?

I wonder if we can make this more of a 'memoized' singleton. i.e. if you request a logger with the exact same url (and other initialization params), you get the same instance.

Does Swift let you return a different instance from `init`? Or would memoization like that require some kind of separate factory function.

Reply:
>if multiple people use the same shared logger and change the url we're in trouble, right?

I am thinking of a use-case of when this might happen :-) chances for this are very very rare

Reply:
@ZkHaider I'm not so sure about having this be a singleton for the reason @bluepnume makes above on possible changes to url and other fields.  If it is a class instance, then he can control required and optional  fields and not assume the user will do that.